hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -515,6 +515,12 @@ object Simplify {
           FastIndexedSeq("row"))
       TableMapRows(te, GetField(Ref("row", te.typ.rowType), "row"))
 
+    case TableDistinct(TableDistinct(child)) => TableDistinct(child)
+    case TableDistinct(TableAggregateByKey(child, expr)) => TableAggregateByKey(child, expr)
+    case TableDistinct(TableMapRows(child, newRow)) => TableMapRows(TableDistinct(child), newRow)
+    case TableDistinct(TableLeftJoinRightDistinct(child, right, root)) => TableLeftJoinRightDistinct(TableDistinct(child), right, root)
+    case TableDistinct(TableRepartition(child, n, strategy)) => TableRepartition(TableDistinct(child), n, strategy)
```

Comment:
This one can result in slightly different partitions being drawn, right? so maybe only valid if we don't care about partitioning.

Reply:
hmmm. you're probably right. I suspect that this does currently affect partitioning, but it probably shouldn't.